src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -73,66 +74,70 @@ .maxstack  2
 }");
         }
 
-        [Fact]
+        [Theory]
         [WorkItem(23358, "https://github.com/dotnet/roslyn/issues/23358")]
-        public void EmptyOrNullArrayConv()
+        [InlineData("byte")]
+        [InlineData("char")]
+        [InlineData("uint")]
+        [InlineData("long")]
+        public void EmptyOrNullArrayConv(string type)
         {
-            var comp = CreateCompilationWithMscorlibAndSpan(@"
+            var comp = CreateCompilationWithMscorlibAndSpan(@$"
 using System;
 
 class Test
-{
+{{
     public static void Main()
-    {       
-        var s1 = (ReadOnlySpan<byte>)new byte[]{};
-        var s2 = (ReadOnlySpan<byte>)new byte[]{};
+    {{       
+        var s1 = (ReadOnlySpan<{type}>)new {type}[]{{}};
+        var s2 = (ReadOnlySpan<{type}>)new {type}[]{{}};
 
         Console.Write(s1.Length == s2.Length);
 
-        s1 = (ReadOnlySpan<byte>)(byte[])null;
-        s2 = (ReadOnlySpan<byte>)(byte[])null;
+        s1 = (ReadOnlySpan<{type}>)({type}[])null;
+        s2 = (ReadOnlySpan<{type}>)({type}[])null;
 
         Console.Write(s1.Length == s2.Length);
-    }
-}
+    }}
+}}" + (type != "byte" ? RuntimeHelpersCreateSpan : ""), TestOptions.ReleaseExe);
 
-", TestOptions.ReleaseExe);
-
-            CompileAndVerify(comp, expectedOutput: "TrueTrue", verify: Verification.Passes).VerifyIL("Test.Main", @"
-{
+            CompileAndVerify(comp,
+                expectedOutput: type == "byte" ? "TrueTrue" : null,
+                verify: type == "byte" ? Verification.Passes : Verification.Skipped).VerifyIL("Test.Main", @$"
+{{
   // Code size       77 (0x4d)
   .maxstack  2
-  .locals init (System.ReadOnlySpan<byte> V_0, //s1
-                System.ReadOnlySpan<byte> V_1, //s2
-                System.ReadOnlySpan<byte> V_2)
+  .locals init (System.ReadOnlySpan<{type}> V_0, //s1
+                System.ReadOnlySpan<{type}> V_1, //s2
+                System.ReadOnlySpan<{type}> V_2)
   IL_0000:  ldloca.s   V_2
-  IL_0002:  initobj    ""System.ReadOnlySpan<byte>""
+  IL_0002:  initobj    ""System.ReadOnlySpan<{type}>""
   IL_0008:  ldloc.2
   IL_0009:  stloc.0
   IL_000a:  ldloca.s   V_2
-  IL_000c:  initobj    ""System.ReadOnlySpan<byte>""
+  IL_000c:  initobj    ""System.ReadOnlySpan<{type}>""
   IL_0012:  ldloc.2
   IL_0013:  stloc.1
   IL_0014:  ldloca.s   V_0
-  IL_0016:  call       ""int System.ReadOnlySpan<byte>.Length.get""
+  IL_0016:  call       ""int System.ReadOnlySpan<{ type}>.Length.get""
```

Comment:
>{ type} [](http://example.com/codeflow?start=49&length=7)

nit: formatting here and two lines below

Reply:
Grr, VS auto-formatting.  Thanks, will fix.